jschon/vocabulary/core.py

```diff
@@ -78,6 +80,14 @@ def __init__(self, parentschema: JSONSchema, value: str):
 
         parentschema.uri = uri
 
+    def _get_uri_validate_kwargs(self):
+        return {'allow_non_empty_fragment': False}
+
+
+class IdKeyword_Next(IdKeyword_2019_09):
```

Comment:
This can be moved to a new `vocabulary.future` module, and should inherit directly from `Keyword` and contain a full, explicit constructor implementation.